### ISSUE TITLE
@extend placeholder don't pass selector to other placeholder in some cases

### ISSUE THREAD

**Issue by ervasive** [sass/sass#932]
Yeah, that sounds kinda weird, could'n come up with better title, sorry guys.

That's not pretty often used technique for sure, I hope though it may throw some light on this behavior for me and hopefully for some other people or catch a bug.

The readability and ease of edit of some aspects of generated CSS is really important in my case, so I have to use SASS not in the most comfortable manner.

So I came up with the following scheme:

Two types of styles: skin, structural

To not repeat colours throughout the skin file, I define some shared placeholders. For example:

``` scss
%text-colour { color: #ccc; }
```

Then, to keep all skin related styles in one place I create so-called compound placeholders, which in its turn may contain references to different shared placeholders as well as implement some unique styles.

![showcase](https://f.cloud.github.com/assets/85254/1199482/067a1ef0-2511-11e3-8eea-3cb2a0e92f4f.png)

The problem arise when one of the compound placeholders refers to a single shared placeholder in its sub hierarchical rules. For example:

``` scss
%compound-placeholder {
  @extends %text-colour; // Firs time we use it here

  // Works fine since extends not previously used placeholder.
  &:hover { @extend %alt-color; }

  // Fail to extend shared placeholder, which was used at the
  // root of this placeholder.
  &:active:hover { @extend %text-colour; }
}
```

Here is the example: http://codepen.io/anon/pen/iwons

Cheers.

**Comment by nex3** [comment]
What precisely is the CSS output you expect, and how does the output you're getting differ from that?

**Comment by ervasive** [comment]
Sorry, probably I wasn't clear enough.
Referring to the [example](http://codepen.io/anon/pen/iwons)  I provided above, the output I expect, should have both selectors, listed at the point where `%text-colour` was declared.

```
.test-case,
.test-case:active {
  color: red;
}

.test-case:hover {
  color: green;
}
```

instead I get:

```
.test-case {
  color: red;
}

.test-case:hover {
  color: green;
}
```

So, the `.test-case:active` is missing from the output.

**Comment by nex3** [comment]
Fixed by 1c7a50b047d4c9856fbf6239d8d5b7fc457e6b23.
